core/src/main/java/io/dekorate/kubernetes/decorator/ApplyImageDecorator.java

```diff
@@ -41,4 +42,19 @@ public void andThenVisit(ContainerFluent container) {
         AddSidecarDecorator.class };
   }
 
+  @Override
+  public String getConfigReference() {
+    return getContainerName() + ".image";
+  }
+
+  @Override
+  public String getJsonPathProperty() {
+    return "$.[?(@.kind == 'Deployment')].spec.template.spec.containers[?(@.name == '" + getContainerName() + "')].image";
```

Comment:
The decorator could be used not only in `Deployment` but in `DeploymentConfig`, knative `Service` and so on.
With this in mind, I think that the json path expression is way too limiting.
Not a json path expert but I think that this might be more appropriate:
`$..containers[?(@.name == '" + getContainerName() +"'].image`

Reply:
the container name should never be null
Also, the decorator seems to be used only for k8s -> I need to investigate more why.
However, following your comment, I made your requested changes.